handler/src/test/java/io/netty/handler/ssl/JdkSslEngineTest.java

```diff
@@ -237,49 +249,80 @@ public void testAlpnNoCompatibleProtocolsClientHandshakeFailure() throws Excepti
+            SslContext clientSslCtx = new JdkSslClientContext(null, InsecureTrustManagerFactory.INSTANCE, null,
+                                                   IdentityCipherSuiteFilter.INSTANCE, clientApn, 0, 0);
+
+            mySetup(serverSslCtx, clientSslCtx);
             assertTrue(clientLatch.await(2, TimeUnit.SECONDS));
             assertTrue(clientException instanceof SSLHandshakeException);
-        } catch (Exception e) {
+        } catch (SkipTestException e) {
             // ALPN availability is dependent on the java version. If ALPN is not available because of
             // java version incompatibility don't fail the test, but instead just skip the test
             assumeNoException(e);
         }
     }
 
-    private void mySetup(JdkApplicationProtocolNegotiator apn) throws InterruptedException, SSLException,
+    private void mySetup(ApplicationProtocolConfig apn) throws InterruptedException, SSLException,
             CertificateException {
         mySetup(apn, apn);
     }
 
-    private void mySetup(JdkApplicationProtocolNegotiator serverApn, JdkApplicationProtocolNegotiator clientApn)
+    private void mySetup(ApplicationProtocolConfig serverApn, ApplicationProtocolConfig clientApn)
             throws InterruptedException, SSLException, CertificateException {
         SelfSignedCertificate ssc = new SelfSignedCertificate();
-        serverSslCtx = new JdkSslServerContext(ssc.certificate(), ssc.privateKey(), null, null,
-                IdentityCipherSuiteFilter.INSTANCE, serverApn, 0, 0);
-        clientSslCtx = new JdkSslClientContext(null, InsecureTrustManagerFactory.INSTANCE, null,
-                IdentityCipherSuiteFilter.INSTANCE, clientApn, 0, 0);
+
+        mySetup(SslContextBuilder.forServer(ssc.certificate(),
+                                            ssc.privateKey(),
+                                            null)
+                                 .sslProvider(sslProvider())
+                                 .ciphers(null, IdentityCipherSuiteFilter.INSTANCE)
+                                 .applicationProtocolConfig(serverApn)
+                                 .sessionCacheSize(0)
+                                 .sessionTimeout(0)
+                                 .build(),
+
+                SslContextBuilder.forClient()
+                                 .sslProvider(sslProvider())
+                                 .applicationProtocolConfig(clientApn)
+                                 .trustManager(InsecureTrustManagerFactory.INSTANCE)
+                                 .ciphers(null, IdentityCipherSuiteFilter.INSTANCE)
+                                 .sessionCacheSize(0)
+                                 .sessionTimeout(0)
+                                 .build());
+    }
+
+    private void mySetup(SslContext serverCtx, SslContext clientCtx)
+      throws InterruptedException, SSLException, CertificateException {
```

Comment:
Oh, sorry, I'm using intellij, configured it to not use aligned indentation, hope new version looks ok.